test/e2e/testdata/extract-event-time-from-payload.yaml

```diff
@@ -3,6 +3,9 @@ kind: Pipeline
 metadata:
   name: extract-event-time
 spec:
+  spec:
+    limits:
+      readBatchSize: 1
```

Comment:
without this did you see any problem?

Reply:
No. I added this to ensure that in E2E test, each input message is treated as a separate batch. Such that each new event time gets used when publishing watermark(not just the first message).

Based on my observation, removing this is also fine. Probably because in e2e test, I wait 2 seconds before sending next message, which automatically makes each message a separate batch.

Reply:
let's not add a 2 sec delay in the data generation for test, it will slow down the test experience. i am fine with this setting, i asked because i was just curious :)

Reply:
Adding 2 sec delay is to mitigate the watermark propagation delay. Without the 2 sec delay, daemon client getWatermark call returns -1.

Reply:
if we add a batch size of 1, will that problem be solved? since we are propagating for every message.

Reply:
Unfortunately no. I am still seeing -1.